Mathlib/RepresentationTheory/Induced.lean

```diff
@@ -0,0 +1,158 @@
+/-
+Copyright (c) 2025 Amelia Livingston. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Amelia Livingston
+-/
+import Mathlib.RepresentationTheory.Coinvariants
+
+/-!
+# Induced representations
+
+Given a commutative ring `k`, a group homomorphism `φ : G →* H`, and a `k`-linear
+`G`-representation `A`, this file introduces the induced representation $Ind_G^H(A)$ of `A` as
+an `H`-representation.
+
+By `ind φ A` we mean the `(k[H] ⊗[k] A)_G` with the `G`-representation on `k[H]` defined by `φ`.
+We define a representation of `H` on this submodule by sending `h : H` and `⟦h₁ ⊗ₜ a⟧` to
+`⟦h₁h⁻¹ ⊗ₜ a⟧`.
+
+We also prove that the restriction functor `Rep k H ⥤ Rep k G` along `φ` is right adjoint to the
+induction functor and hence that the induction functor preserves colimits.
+
+## Main definitions
+
+* `Representation.ind φ ρ` : given a group homomorphism `φ : G →* H`, this is the induction of a
+  `G`-representation `(A, ρ)` along `φ`, defined as `(k[H] ⊗[k] A)_G` and with `H`-action given by
+  `h • ⟦h₁ ⊗ₜ a⟧ := ⟦h₁h⁻¹ ⊗ₜ a⟧` for `h, h₁ : H`, `a : A`.
+* `Rep.indResAdjunction k φ`: given a group homomorphism `φ : G →* H`, this is the adjunction
+  between the induction functor along `φ` and the restriction functor `Rep k H ⥤ Rep k G`
+  along `φ`.
+
+-/
+
+universe u
+
+namespace Representation
+
+open Finsupp TensorProduct
+
+variable {k G H : Type*} [CommRing k] [Group G] [Group H] (φ : G →* H) {A B : Type*}
+  [AddCommGroup A] [Module k A] (ρ : Representation k G A)
+  [AddCommGroup B] [Module k B] (τ : Representation k G B)
+
+/-- Given a group homomorphism `φ : G →* H` and a `G`-representation `(A, ρ)`, this is
+`(k[H] ⊗[k] A)_G` with the `G`-representation on `k[H]` defined by `φ`. -/
```

Comment:
The following is just a suggestion. Does it match what you intended?
```suggestion
/-- Given a group homomorphism `φ : G →* H` and a `G`-representation `(A, ρ)`, this is is the underlying module of
`(k[H] ⊗[k] A)_G` with the `G`-representation on `k[H]` defined by `φ`.

See `Representation.ind` for the corresponding representation. -/
```

Reply:
Pretty much - thank you!